### ISSUE TITLE
XCode 12.5 Support critical due to Visual Studio stable update

### ISSUE THREAD

**Issue by buzzware** [microsoft/appcenter#2185]
**Describe the solution you'd like**
To be able to select XCode 12.5 for builds

**Describe alternatives you've considered**
My VS for Mac updated without me realising it would require XCode 12.5 which would require updating MacOS to Big Sur.
I looked for how to downgrade Xamarin.iOS to continue with XCode 12.4 and haven't found a good solution, or what version I should get.
I have now upgraded my dev machine in hope and faith that you will add XCode 12.5 before I really need it.
The only alternative is to find another CI/CD provider.

**Additional context**

**Comment by bwised** [comment]
I'm really surprised that we have to make this request ourselves in the first place.  One of the primary purposes of using a service like AppCenter in the first place is so that these kind of basic essential things get taken care of.  We pay for this service!

**Comment by buzzware** [comment]
> I was told by a support person that I had to come on this damn issue posting and "thumbs up" this if I wanted to see it get done. This is rediculous. We pay for this service -- this isn't some feature request or bug -- it's essential for the service to even be useful!

Yes, this is not optional, and while it turns out my my project builds on both 12.4 and 12.5, that could change because of some package update or some other project.
Perhaps part of the reason for the delay is because they have to upgrade to Big Sur, but yes, they had access to the beta.

**Comment by beeradmoore** [comment]
I don't have anything preventing me from deploying on 12.4, but it is still a pain that it is not updated which is a problem for others. (Assuming Apple still let us submit with the older SDK?) 

The [roadmap](https://github.com/microsoft/appcenter/wiki/Roadmap) says 
> Framework and OS updates will continue to be supported in a timely manner.

🤷‍♂️

**Comment by boxtcarpel** [comment]
I too was asked by support to post on this issue in order to request Xcode 12.5 support, after being told that "At this time there are no plans to update support for Xcode 12.5.".
This does not make sense for a paid build system people rely on.

Additionally, their documentation at https://docs.microsoft.com/en-us/appcenter/build/ios/ state:
"Supported versions and requirements
Build machine Xcode version details are updated each time a new version of Xcode is added. We keep an eye on the latest versions released by Apple and include them as soon as possible on the VMs used to run the builds."

Xcode 12.5 (https://developer.apple.com/news/releases/?id=04262021f) came out two weeks ago at this point, and betas were around long before that.

This is bordering on unacceptable, and would cross that border if it repeats with future versions...

**Comment by beeradmoore** [comment]
> At this time there are no plans to update support for Xcode 12.5.

Yikes. Yeah, going to start looking elsewhere I think.

**Comment by wisdeom** [comment]
Hi all, Xcode 12.5 requires mac OS Big Sur 11, and we have an external dependency to complete this support. We're closely working with Azure DevOps team to have mac OS 11 image to get this done, and this is their 1st priority now. 
https://github.com/actions/virtual-environments/issues/2486

We'll keep you posted on the progress. Sorry for any inconvenience caused.

**Comment by JetForMe** [comment]
This is a big issue for us, too. All the build services providers seem to have been caught with their pants down around macOS 11, and while I agree Apple's terms are onerous, other providers are trialing solutions. In the meantime we're rolling our own using Xcode server.

**Comment by anfriis** [comment]
Big issue here as well. 
There is a bug in the `ipatool` in Xcode **12.4** that fails in processing the xcframework's we have in our app, so we are not able to build and distribute the app. 

The bug is fixed in Xcode 12.5 which is able to complete the processing and create an ipa file on our local machine. 
Hoping for an update soon

**Comment by joemather** [comment]
@wisdeom - Could we please get an update on this issue?  

The macOS 11 image is available and Xcode 12.5 is supported:
https://github.com/actions/virtual-environments/issues/2486#issuecomment-854245059

Xcode 12.5 will be the default starting June 15:
https://github.com/actions/virtual-environments/issues/3522

Thanks.

**Comment by buzzware** [comment]
Heads up Azure & AppCenter people: macOS 12 Monterey is available now.
Xcode will become dependent on it too one day - may as well start the process early so this issue doesn't repeat.
p.s. We shouldn't have to remind you.

**Comment by douglasjunior** [comment]
> Although there's no problem for most of my apps being built with Xcode 12.4. One of my apps is being affected, its building will always fail due to bitcode version unsupported. As mentioned AppCenter is not a free service, instead, we pay for it. At least we should be noticed when will this support be planned.

Same here, an app that uses the recent Firebase iOS SDK.

**Comment by buzzware** [comment]
Any company or team trying to instill quality production practices would encourage or require a standard environment for development, CI & CD. If they keep their development laptops up to date with the latest OSX Big Sur (released 7 months ago) or certain packages up to date they are now forced to XCode 12.5 and if they have been using AppCenter they are now forced to have an inconsistent environment by a service they likely pay for, from the provider of their development environment (Xamarin), and one of the largest few tech companies. 

This is a very surprising, frustrating and unusual scenario. The responses so far have been very disappointing.

**Comment by beeradmoore** [comment]
@buzzware, yep. It seems AppCenter and Azure Devops/Pipeline all use the same system as Github Actions in the backend. They are the ones who hadn't updated to the latest macOS version over the past 12 months. Not sure why that is the case, but I am sure they have their reasons. 

On the bright side Xcode 12.5 is the default build SDK on macOS-11 in GitHub actions starting [today](https://github.com/actions/virtual-environments/issues/3522) (or sometime over the next few days). But that is still preview. Another thing I spotted was Xcode 12.2 and 12.3 is [being removed](https://github.com/actions/virtual-environments/issues/3555) on the 28th so if you weren't already using 12.4 you now will need to swap to that.

I can't see any word when we can expect macOS-11 to go public to everyone.
I can't see any word from AppCenter on if they got into the macOS-11 preview.
I can't see any word from AppCenter that their systems are working with macOS-11 preview as expected.
I can't see any word form AppCenter when Xamarin.Android can be updated to 11.2.

Looking for an alternative solution (unless I missed something) it seems the only working solution today is to roll with your own Jenkins build server. We moved away from that to AppCenter because it was cheaper to pay for that than it was pay someone to upkeep the build server.

Hopefully we can hear some good news soon.

**Comment by douglasjunior** [comment]
AppCenter's iOS SDK [updated to Xcode 12.5 yesterday](https://github.com/microsoft/appcenter-sdk-apple/releases/tag/4.2.0).

**Comment by douglasjunior** [comment]
Response from AppCenter support team.

> Our team has been already working on Xcode 12.5 support. It's planned at the beginning of May. Currently, the status is that we need to wait for Azure DevOps pool to add macOS Big Sur (macOS 11) as Xcode 12.5 has dependency to have Big Sur 11. App Center Build is based on Azure DevOps, so it's blocked at this time.

In short, no release date at the moment.

**Comment by beeradmoore** [comment]
> AppCenter's iOS SDK [updated to Xcode 12.5 yesterday](https://github.com/microsoft/appcenter-sdk-apple/releases/tag/4.2.0).

That's awkward 👀


> Response from AppCenter support team.

Thank you for updating us. Would have been nice if AppCenter would have told us this themselves.

**Comment by Watersdr** [comment]
Here we are almost 2 months from "the beginning of May" still waiting...

**Comment by cquanu** [comment]
Any updates?

**Comment by neoswallow** [comment]
Since Azure Pipelines has macOS-11 as private preview, is there such program to join for App Center as well for Xcode 12.5 support?

**Comment by andrew-lundy** [comment]
The App Center SDK is separate from the App Center Build service.

The build service doesn't support Xcode 12.5 because there are no build agents that run macOS 11, [which is needed to run Xcode 12.5](https://developer.apple.com/documentation/xcode-release-notes/xcode-12_5-release-notes). Because of this, adding support for Xcode 12.5 within the App Center Build service is blocked. That said, the App Center product team is working closely with the Azure DevOps team in order to get the build agents needed on App Center. You can keep your eye on this process [here](https://github.com/actions/virtual-environments/issues/2486).


Right now, the App Center SDK supports Xcode 12.5, as seen in release [4.2.0](https://github.com/microsoft/appcenter-sdk-apple/releases/tag/4.2.0). This means that users can build with the SDK on their local machine, as long as they have a machine that runs macOS 11.

**Comment by cquanu** [comment]
When can we get Xcode version 12.5 for our build configuration in the App Center?

**Comment by christophemenager** [comment]
I really don't understand the position of appcenter towards its customers. 
We have no ETA at all, no official communication. 
It feels like they don't care about loosing customers, so that's what I am about to do : there are many competitive alternative on this market

**Comment by beeradmoore** [comment]
It's because AppCenter can't give an ETA, it's out of their hands. They are still waiting on other teams, and other teams teams.

**Comment by cquanu** [comment]
I come here every day to look for updates. But the team of App Center simply don't care but was able to mark one of my new issue as duplicate. Amazing.

**Comment by ThanhNg** [comment]
They actually had no plan and ETA for this, I mean at all. Xamarin and AppCenter are dying...

**Comment by cquanu** [comment]
By the time iOS 15 is released, we still don't have support and I have to build my app on XCode instead. 🤣

**Comment by akamenev** [comment]
Hi all, we understand that this is an important update and we are sorry that we do not have it yet. As was mentioned previously on this thread, App Center Build is dependant on Azure Pipelines and that's why we cannot support it until Big Sur is generally available there. Please follow [this issue](https://github.com/actions/virtual-environments/issues/2072) for the most recent updates from the Azure DevOps team.

**Comment by devshorts** [comment]
That's a garbage excuse, the whole point of using a hosted CICD is to stay up to date. If you guys are going to be months behind the curve on what are often times _forced_ updates for ios and osx developers then we might as well all switch off appcenter and use something else. I can't believe the disdain you have for your customers!

**Comment by cquanu** [comment]
> Because when ios 15 drops everyone will be forced to find alternatives...

Exactly the point here. Once iOS 15 is out. There's no need for me to stay on AppCenter if it can't build for iOS 15.

This is completely ridiculous￼, when iOS 15 is out we're 2 image (macOS) behind.

Big Sur and Monterey.

**Comment by s-hintz** [comment]
Our builds do fail because a 3rd party SDK only works with Xcode 12.5 but not with Xcode 12.4.

ANOMIDImplementation.h:18:9: fatal error: module 'OMSDK_Appnexus' not found
@import OMSDK_Appnexus;
 ~~~~~~~^~~~~~~~~~~~~~

**Comment by s-hintz** [comment]
**Just a reminder:**
Big Sur was announced on June 22, 2020, and was released to the public on November 12, 2020.

Does Microsoft really need about one year to install an update?
Is anybody working on it or do you want us to say good bye?

**Comment by jamesone** [comment]
Can this be prioritised by the azure team somehow? Can one of the Appcenter collaborators get in contact with the Azure team and get this prioritised? 

Appcenter is awesome but if they continue to be behind the 8ball when it comes to ensuring the most up to date xcode is available we may have to look elsewhere as it'll cause many delays that will end up costing $$$$.

**Comment by cquanu** [comment]
Any updates @akamenev?

**Comment by anhar** [comment]
Any updates regarding this? I'm currently having to ask my team to downgrade back to Xcode 12.4 and frankly it doesn't feel right.

**Comment by cquanu** [comment]
Following the announcement here https://github.com/actions/virtual-environments/issues/2486#issuecomment-899808338

macOS-11 is out of preview and now generally available for all the GitHub customers 🥳

And we still don't have support for AppCenter. Is there no SLA?

**Comment by JustinKent** [comment]
At your earliest convenience, please add Xcode 12.5 support to AppCenter build pipelines. Thank you in advance.

**Comment by cquanu** [comment]
Any updates?

**Comment by derekblank** [comment]
This is very concerning for the future of App Center. Even if 12.5 was supported tomorrow, App Center is still behind on supporting the Xcode 13 betas that have been out for months.

Our team is interpreting this as a signal that Microsoft does not intend to support App Center long-term, and are preparing to move to another build service. That will be an expensive headache, especially since it is avoidable, but this issue has cost us thousands of dollars already. It should not matter if there is an issue with GitHub Actions. We are paying to use _App Center_, and App Center should be keeping its documentation up to date and be transparent with support plans. I could understand this issue if it were an open source tool with a single, overworked maintainer, but it is quite shocking to see this coming from an enterprise-level paid service. 

@akamenev @vijayma At the very least, it will help other developers if you are able to remove the misleading statements on the App Center documentation regarding [supported versions and requirements](https://docs.microsoft.com/en-us/appcenter/build/ios/#supported-versions-and-requirements): 

> Build machine Xcode version details are updated each time a new version of Xcode is added. We keep an eye on the latest versions released by Apple and include them as soon as possible on the VMs used to run the builds.

**Comment by will-gluwa** [comment]
In https://github.com/actions/virtual-environments/issues/2072, @maxim-lobanov said:
>Note: Availability of macOS 11 in Azure DevOps doesn't mean that the new image will be available in App Center immediately. App Center team has to perform some work on their side to make sure that new Xcode versions are available in App Center Build.

Any details on when this might be ready? If it's not solved soon I'll have to try setting up builds in Azure DevOps directly

**Comment by douglasjunior** [comment]
> That will be an expensive headache, especially since it is avoidable, but this issue has cost us thousands of dollars already. 

Same here @derekblank, we have about 20 apps that we need to move to another build service.

**Comment by Watersdr** [comment]
> Hi all, we understand that this is an important update and we are sorry that we do not have it yet. As was mentioned previously on this thread, App Center Build is dependant on Azure Pipelines and that's why we cannot support it until Big Sur is generally available there. Please follow [this issue](https://github.com/actions/virtual-environments/issues/2072) for the most recent updates from the Azure DevOps team.

According to the latest comment in the issue linked here, they should no longer be blocked as of 5 hours ago when writing this comment 👀

**Comment by JustinKent** [comment]
Hang in there! Looks like we’ll need a little more patience:

> Note: Availability of macOS 11 in Azure DevOps doesn't mean that the new image will be available in App Center immediately. App Center team has to perform some work on their side to make sure that new Xcode versions are available in App Center Build.

**Comment by cquanu** [comment]
@akamenev @vijayma any plans for AppCenter?

> Note: Availability of macOS 11 in Azure DevOps doesn't mean that the new image will be available in App Center immediately. App Center team has to perform some work on their side to make sure that new Xcode versions are available in App Center Build.

**Comment by andrew-lundy** [comment]
Here is an update from the Azure team: https://github.com/actions/virtual-environments/issues/2072#issuecomment-904371992.

> Availability of macOS 11 in Azure DevOps doesn't mean that the new image will be available in App Center immediately. App Center team has to perform some work on their side to make sure that new Xcode versions are available in App Center Build.

**Comment by cquanu** [comment]
> Here is an update from the Azure team: [actions/virtual-environments#2072 (comment)](https://github.com/actions/virtual-environments/issues/2072#issuecomment-904371992).
> 
> > Availability of macOS 11 in Azure DevOps doesn't mean that the new image will be available in App Center immediately. App Center team has to perform some work on their side to make sure that new Xcode versions are available in App Center Build.

This update says nothing about what the AppCenter team will do.

**Comment by wisdeom** [comment]
Closing this as App Center Build now fully supports Xcode 12.5.

**Comment by buzzware** [comment]
Oh great, congratulations to any internal heroes that got this done.
Should we raise another issue for MacOS 12 now, so its ready when we need it? You can get the beta.

**Comment by beeradmoore** [comment]
@buzzware, Looks like the other teams that need to work their magic before AppCenter can do theirs are [aware of it](https://github.com/actions/virtual-environments/issues/3649).
